### PR TITLE
Reorg the examples with CMake command flows like in HDF5

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: HDF4Examples/C/AN/C_sourcefiles.cmake [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
 #-----------------------------------------------------------------------------
 set (h4examples
-    h4ex_AN_create_annotation.c
+    AN_create_annotation
 )
 set (h4examples_use
-    h4ex_AN_get_annotation_info.c
-    h4ex_AN_read_annotation.c
+    AN_get_annotation_info
+    AN_read_annotation
```
Comment (bmribler) [inline HDF4Examples/C/AN/C_sourcefiles.cmake:9]: These don't need some sort of H4/h4 prefix?

Comment (byrnHDF) [reply HDF4Examples/C/AN/C_sourcefiles.cmake:9]: No, that gets added in the CMakeLists.txt, simplified it.

File: HDF4Examples/CMakeLists.txt [hunk 2]
```diff
@@ -4,29 +4,41 @@ project (H4EXAMPLES C)
...
-set (HDF4EX_JAVA_DIR            ${H4EXAMPLES_SOURCE_DIR}/JAVA)
-set (HDF4EX_RESOURCES_DIR       ${H4EXAMPLES_SOURCE_DIR}/config/cmake)
+set (H4EX_C_DIR              ${H4EXAMPLES_SOURCE_DIR}/C)
+set (H4EX_HDF_FORTRAN_DIR    ${H4EXAMPLES_SOURCE_DIR}/hdf/FORTRAN)
+set (H4EX_MFHDF_FORTRAN_DIR  ${H4EXAMPLES_SOURCE_DIR}/mfhdf/FORTRAN)
+set (H4EX_JAVA_DIR           ${H4EXAMPLES_SOURCE_DIR}/JAVA)
 
 #-----------------------------------------------------------------------------
-# Basic HDF4Examples stuff here
+# Basic HDF5Examples stuff here
```
Comment (lrknox) [inline HDF4Examples/CMakeLists.txt:13]: This looks wrong?

**Review by lrknox — APPROVED** [review]

File: HDF4Examples/CMakeLists.txt (hunk 2 above)
Comment (byrnHDF) [reply HDF4Examples/CMakeLists.txt]: Fixed

**Pull request merged**